source/adapters/opencl/platform.cpp

```diff
@@ -96,6 +96,18 @@ urPlatformGet(ur_adapter_handle_t *, uint32_t, uint32_t NumEntries,
     }
   }
 
+  if (Result == CL_INVALID_VALUE) {
+    // From OpenCL:
+    // if num_entries is equal to zero and platforms is not NULL or if both
+    // num_platforms and platforms are NULL.
+    if (NumEntries == 0 && phPlatforms != nullptr) {
```

Comment:
Do we really need to repeat this check if it is already happening within validation layer?

https://github.com/oneapi-src/unified-runtime/blob/c742ca49efb12380a35b8b0b467e6577ab8174ce/source/loader/layers/validation/ur_valddi.cpp#L234-L236

I'm asking mostly out of curiosity, because I don't know what is the error handling policy within UR

Reply:
We don't enable the validation layer from the sycl RT, although I think that decision was made while we were trying to fix a performance regression that later turned out to be something totally unrelated. I don't like having this check twice, maybe we should look to see what the perf impact of enabling just basic parameter validation would be

Reply:
The validation layer is opt-in and is disabled by default in the SYCL RT. If checks are required for correctness based on SYCL RT usage patterns then they need to exist in the adapter.

Reply:
> The validation layer is opt-in and is disabled by default in the SYCL RT. If checks are required for correctness based on SYCL RT usage patterns then they need to exist in the adapter.

https://github.com/intel/llvm/blob/sycl/sycl/source/detail/adapter.hpp#L103 - if you look at the code here, we do not have a check that `platformCount` returned to SYCL RT is 0. [`std::vector::data`](https://en.cppreference.com/w/cpp/container/vector/data) does not guarantee that `nullptr` would be returned for an empty vector, so theoretically we _can_ encounter a case that we check in the adapter. However, in my local testing I wasn't able to encounter it, because `vector::data` always returned `nullptr` (empty `vector` resized to `0` elements).

I think that we should definitely update the SYCL RT code here to avoid performing an incorrect call and therefore it should allow to simplify the code in the adapter.

Reply:
I've added that check now in the associated intel/llvm change.